.github/workflows/android.yml

```diff
@@ -1,18 +1,16 @@
 name: Android CI
 
-on:
-  push:
-    branches: [ main ]
-  # Build on all pull requests, regardless of target.
-  pull_request:
+on :
+  pull_request :
+  merge_group :
```

Comment:
Does this need to be changed as part of this?

Reply:
It depends on how securely we want to lock down the custom runner. if we specify the workflows it can run on rather than just open it to any workflow then we need a stable workflow name - which can only happen if we run it on the `main` branch rather than the new branch that is created for each PR.